fault_tolerance_slave.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+"""
+Copyright 2019 Marco Lattuada
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import os
+import sys
+
+import sequence_data_processing
+
+
+def generate_test():
+    test = {
+        'General':{
+            'run_num': 1,
+            'techniques': ['LRRidge','DecisionTree','XGBoost'],
+            'hp_selection': 'KFold',
+            'validation': 'HoldOut',
+            'folds': 4,
+            'hold_out_ratio': 0.2,
+            'y': 'ave_response_time',
+            'hyperparameter_tuning': 'Hyperopt',
+            'hyperopt_max_evals': 10,
+            'hyperopt_save_interval': 5
+        },
+        'DataPreparation':{
+            'input_path': 'inputs/faas_test.csv',
+            'inverse': ['Lambda']
+        },
+        'FeatureSelection':{
+            'method': 'SFS',
+            'max_features': 3,
+            'folds': 3
+        },
+        'LRRidge':{
+            'alpha': ['loguniform(0.01,1)']
+        },
+        'DecisionTree':{
+            'criterion': ['mse'],
+            'max_depth': [2,3,5],
+            'max_features': ['auto'],
+            'min_samples_split': [0.01,0.1,0.5],
+            'min_samples_leaf': [0.01,0.1,0.5]
```

Comment:
I would use a smaller grid, by removing the second entries of the parameters with three